projects/schematics/src/add-cms-component/index.ts

```diff
@@ -0,0 +1,390 @@
+      CMS_COMPONENT_DATA_PROPERTY_NAME
+    );
+    changes.push(injectionChange);
+
+    const componentDataProperty = `  ${CMS_COMPONENT_DATA_PROPERTY_NAME}$: Observable<${strings.classify(
+      options.cmsComponentDataModel
+    )}> = this.${CMS_COMPONENT_DATA_PROPERTY_NAME}.data$;`;
+    const componentDataPropertyChange = defineProperty(
+      nodes,
+      componentPath,
+      componentDataProperty
+    );
+    changes.push(componentDataPropertyChange);
+
+    const cmsComponentImport = insertImport(
+      componentTs,
+      componentPath,
+      strings.classify(options.cmsComponentDataModel),
+      options.cmsComponentDataModelPath,
+      false
+    );
+    changes.push(cmsComponentImport);
+
+    const cmsComponentDataImport = insertImport(
+      componentTs,
+      componentPath,
+      CMS_COMPONENT_DATA_CLASS,
+      SPARTACUS_STOREFRONTLIB,
+      false
+    );
+    changes.push(cmsComponentDataImport);
+
+    const observableImport = insertImport(
+      componentTs,
+      componentPath,
+      OBSERVABLE_CLASS,
+      RXJS,
+      false
+    );
+    changes.push(observableImport);
+
+    commitChanges(tree, componentPath, changes, InsertDirection.LEFT);
+  };
+}
+
+function updateTemplate(options: CxCmsComponentSchema): Rule {
+  return (tree: Tree, _context: SchematicContext) => {
+    const componentFileName = `${strings.dasherize(
+      options.name
+    )}.${strings.dasherize(options.type)}.ts`;
+    const possibleProjectFiles = ['/angular.json', '/.angular.json'];
+    const { workspace } = getWorkspace(tree, possibleProjectFiles);
+    const project = getProjectFromWorkspace(workspace, options.project);
+
+    const componentPath = getPathResultsForFile(
+      tree,
+      componentFileName,
+      project.sourceRoot
+    )[0];
+    const componentTs = getTsSourceFile(tree, componentPath);
+
+    let templatePath = '';
+    let templateContent = '';
+    let startIndex: number;
+    if (options.inlineTemplate) {
+      templatePath = componentPath;
+      const decorator = getDecoratorMetadata(
+        componentTs,
+        'Component',
+        ANGULAR_CORE
+      )[0];
+
+      const inlineTemplate = getMetadataProperty(decorator, 'template');
+      templateContent = inlineTemplate.getText();
+      startIndex = inlineTemplate.name.parent.end - 1;
+    } else {
+      const componentTemplateFileName = `${strings.dasherize(
+        options.name
+      )}.${strings.dasherize(options.type)}.html`;
+      templatePath = getPathResultsForFile(
+        tree,
+        componentTemplateFileName,
+        project.sourceRoot
+      )[0];
+      const buffer = tree.read(templatePath);
+      templateContent = buffer ? buffer.toString(UTF_8) : '';
+      startIndex = templateContent.length;
+    }
+
+    if (Boolean(templateContent)) {
+      const insertion =
+        `  <ng-container *ngIf="${CMS_COMPONENT_DATA_PROPERTY_NAME}$ | async as data">` +
+        `\n      {{data | json}}` +
+        `\n    </ng-container>`;
```

Comment:
When component is created, result html has weird intendation.

Reply:
Is there an API that can return us the existing indentation level?

Reply:
Scratch that, I decided to put everything in one line